ratecoeff.cc

```diff
@@ -76,14 +78,42 @@ static bool read_ratecoeff_dat(void)
       fileisamatch = false;
     }
 
-    char phixsfile_hash_in[33];
-    assert_always(fscanf(ratecoeff_file, "%32s\n", phixsfile_hash_in) == 1);
-    printout("ratecoeff.dat: MD5 %s = %s ", phixsdata_filenames[phixs_file_version], phixsfile_hash_in);
-    if (strcmp(phixsfile_hash, phixsfile_hash_in) == 0)
-      printout("(pass)\n");
-    else {
-      printout("\nMISMATCH: MD5 %s = %s\n", phixsdata_filenames[phixs_file_version], phixsfile_hash);
-      fileisamatch = false;
+    const bool phixs_v1_exists = std::ifstream(phixsdata_filenames[1]).good();
+    const bool phixs_v2_exists = std::ifstream(phixsdata_filenames[2]).good();
+    if (phixs_v1_exists && phixs_v2_exists) {
+      char phixsfile_v2_hash_in[33];
+      char phixsfile_v1_hash_in[33];
+      assert_always(fscanf(ratecoeff_file, "%32s\n", phixsfile_v2_hash_in) == 1);
+      assert_always(fscanf(ratecoeff_file, "%32s\n", phixsfile_v1_hash_in) == 1);
+      printout("ratecoeff.dat: MD5 %s = %s ", phixsdata_filenames[2], phixsfile_v2_hash_in);
+      printout("ratecoeff.dat: MD5 %s = %s ", phixsdata_filenames[1], phixsfile_v1_hash_in);
+      if (strcmp(phixsfile_v2_hash, phixsfile_v2_hash_in) == 0 && strcmp(phixsfile_v1_hash, phixsfile_v1_hash_in) == 0)
+        printout("(pass)\n");
+      else {
+        printout("\nMISMATCH: MD5 %s = %s and MD5 %s = %s\n", phixsdata_filenames[2], phixsfile_v2_hash,
+                 phixsdata_filenames[1], phixsfile_v1_hash);
+        fileisamatch = false;
+      }
+    } else if (phixs_v1_exists) {
+      char phixsfile_v1_hash_in[33];
+      assert_always(fscanf(ratecoeff_file, "%32s\n", phixsfile_v1_hash_in) == 1);
+      printout("ratecoeff.dat: MD5 %s = %s ", phixsdata_filenames[1], phixsfile_v1_hash_in);
+      if (strcmp(phixsfile_v1_hash, phixsfile_v1_hash_in) == 0)
+        printout("(pass)\n");
+      else {
+        printout("\nMISMATCH: MD5 %s = %s\n", phixsdata_filenames[1], phixsfile_v1_hash);
+        fileisamatch = false;
+      }
+    } else {
+      char phixsfile_v2_hash_in[33];
+      assert_always(fscanf(ratecoeff_file, "%32s\n", phixsfile_v2_hash_in) == 1);
+      printout("ratecoeff.dat: MD5 %s = %s ", phixsdata_filenames[2], phixsfile_v2_hash_in);
+      if (strcmp(phixsfile_v2_hash, phixsfile_v2_hash_in) == 0)
+        printout("(pass)\n");
+      else {
+        printout("\nMISMATCH: MD5 %s = %s\n", phixsdata_filenames[2], phixsfile_v2_hash);
+        fileisamatch = false;
+      }
```

Comment:
It looks like you've copied and pasted code. Don't repeat yourself and keep the code as simple as possible, for example:
```
for (int phixsver = 1; phixsver <= 2; phixsver++) {
    if (std::ifstream(phixsdata_filenames[phixsver]).good())) {
        char phixsfile_hash_in[33]
        assert_always(fscanf(ratecoeff_file, "%32s\n", phixsfile_hash_in) == 1);
        printout("ratecoeff.dat: MD5 %s = %s ", phixsdata_filenames[phixsver], phixsfile_hash_in);
        if (strcmp(phixsfile_hash[phixsver], phixsfile_hash_in) == 0) {
            printout("(pass)\n");
        } else {
           printout("\nMISMATCH: MD5 %s = %s\n", phixsdata_filenames[phixsver], phixsfile_hash[phixsver]);
           fileisamatch = false;
        }
    }
}
```